### ISSUE TITLE
[Question] Reports-sidecar cannot be connected to in Podman discovery platform

### ISSUE THREAD

**Issue by maxcao13** [cryostatio/cryostat#1455]
I notice that if we include the Podman discovery, the cryostat-reports container is labelled and is scheduled to be discovered but is always cannot be connected to. What should happen here?

```
INFO: Connection for service:jmx:remote+http://localhost:10000 closed
Apr 17, 2023 9:02:17 PM io.cryostat.core.log.Logger error
SEVERE: Exception thrown
org.openjdk.jmc.rjmx.ConnectionException caused by java.net.MalformedURLException: Unsupported protocol: remote+http
	at org.openjdk.jmc.rjmx.internal.RJMXConnection.connect(RJMXConnection.java:345)
	at io.cryostat.core.net.JFRJMXConnection.attemptConnect(JFRJMXConnection.java:443)
	at io.cryostat.core.net.JFRJMXConnection.connect(JFRJMXConnection.java:399)
	at io.cryostat.core.net.JFRJMXConnection.getJvmId(JFRJMXConnection.java:204)
	at io.cryostat.net.TargetConnectionManager.lambda$executeConnectedTaskAsync$2(TargetConnectionManager.java:153)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.helpAsyncBlocker(ForkJoinPool.java:1253)
	at java.base/java.util.concurrent.ForkJoinPool.helpAsyncBlocker(ForkJoinPool.java:2237)
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1933)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
	at io.cryostat.recordings.JvmIdHelper.resolveId(JvmIdHelper.java:158)
	at io.cryostat.discovery.DiscoveryStorage.modifyChildrenWithJvmIds(DiscoveryStorage.java:317)
	at io.cryostat.discovery.DiscoveryStorage.modifyChildrenWithJvmIds(DiscoveryStorage.java:336)
	at io.cryostat.discovery.DiscoveryStorage.update(DiscoveryStorage.java:348)
	at io.cryostat.discovery.BuiltInDiscovery.lambda$start$2(BuiltInDiscovery.java:140)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1428)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.net.MalformedURLException: Unsupported protocol: remote+http
	at java.management/javax.management.remote.JMXConnectorFactory.newJMXConnector(JMXConnectorFactory.java:366)
	at org.openjdk.jmc.rjmx.internal.RJMXConnection.connectJmxConnector(RJMXConnection.java:570)
	at org.openjdk.jmc.rjmx.internal.RJMXConnection.establishConnection(RJMXConnection.java:552)
	at org.openjdk.jmc.rjmx.internal.RJMXConnection.connect(RJMXConnection.java:338)
	... 22 more
```

**Comment by andrewazores** [comment]
That looks like a mistake in the smoketest script - it's trying to use `remote+http`, which is the same as the Wildfly sample app because it's actually the JBoss remoting connector, which is not implemented by the base JDK of the report sidecar. I'll include a commit in #1449 for this.

**Comment by maxcao13** [comment]
Fixed in #1466
